### PR TITLE
Support async

### LINKED ISSUE #22: Support async test functions
@tworec Saw you'd started looking at this. Is this something you intend to continue with, or should I pick this up?

### PR DESCRIPTION
**Pull request by palfrey**
Fixes #22

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -25,15 +25,7 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
...
-version = "0.5.5"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-dependencies = [
- "loom 0.3.4 (registry+https://github.com/rust-lang/crates.io-index)",
-]
-
-[[package]]
-name = "cc"
-version = "1.0.54"
+version = "0.5.6"
```
Comment (tworec) [inline Cargo.lock:28]: is it manual action or some `cargo`'s neat subcommand?

Comment (palfrey) [reply Cargo.lock:28]: I'm now trying to remember what I did. I suspect I reset the Cargo.lock back to master version and then re-ran `cargo build` or similar.

File: serial_test_derive/src/lib.rs [hunk 2]
```diff
@@ -169,3 +190,41 @@ fn test_stripped_attributes() {
     };
     assert_eq!(format!("{}", compare), format!("{}", stream));
 }
+
+#[test]
+fn test_serial_async() {
+    let attrs = proc_macro2::TokenStream::new();
+    let input = quote! {
+        #[test]
```
Comment (tworec) [inline serial_test_derive/src/lib.rs:198]:
```suggestion
        #[tokio::test]
```

File: serial_test_derive/src/lib.rs [hunk 3]
```diff
@@ -169,3 +190,41 @@ fn test_stripped_attributes() {
...
+        async fn foo() {}
+    };
+    let stream = serial_core(attrs.into(), input);
+    let compare = quote! {
+        #[test]
```
Comment (tworec) [inline serial_test_derive/src/lib.rs:203]:
```suggestion
        #[tokio::test]
```

File: serial_test_derive/src/lib.rs [hunk 4]
```diff
@@ -169,3 +190,41 @@ fn test_stripped_attributes() {
...
+        }
+    };
+    assert_eq!(format!("{}", compare), format!("{}", stream));
+}
+
+#[test]
+fn test_serial_async_return() {
+    let attrs = proc_macro2::TokenStream::new();
+    let input = quote! {
+        #[test]
```
Comment (tworec) [inline serial_test_derive/src/lib.rs:217]:
```suggestion
        #[actix_rt::test]
```

File: serial_test_derive/src/lib.rs [hunk 5]
```diff
@@ -169,3 +190,41 @@ fn test_stripped_attributes() {
...
+        async fn foo() -> Result<(), ()> { Ok(()) }
+    };
+    let stream = serial_core(attrs.into(), input);
+    let compare = quote! {
+        #[test]
```
Comment (tworec) [inline serial_test_derive/src/lib.rs:222]:
```suggestion
        #[actix_rt::test]
```

**Review by tworec — CHANGES_REQUESTED** [review]

File: Cargo.lock (hunk 1 above)
Comment (tworec) [reply Cargo.lock:28]: ;)

**Comment by tworec** [comment]
tested within https://github.com/golemfactory/yagna/pull/497 and seems to work

File: README.md [hunk 6]
```diff
@@ -4,7 +4,7 @@
 [![Docs](https://docs.rs/serial_test/badge.svg)](https://docs.rs/serial_test/)
 [![MIT license](https://img.shields.io/crates/l/serial_test.svg)](./LICENSE)
 [![Build Status](https://travis-ci.com/palfrey/serial_test.svg?branch=master)](https://travis-ci.com/palfrey/serial_test)
-[![MSRV: 1.36.0](https://flat.badgen.net/badge/MSRV/1.36.0/purple)](https://blog.rust-lang.org/2019/07/04/Rust-1.36.0.html)
+[![MSRV: 1.39.0](https://flat.badgen.net/badge/MSRV/1.39.0/purple)](https://blog.rust-lang.org/2019/11/07/Rust-1.39.0.html)
```
[the remark this answers is not in the document]
Comment (tworec) [reply README.md:7]: nope

File: serial_test_derive/src/lib.rs (hunk 4 above)
Comment (tworec) [reply serial_test_derive/src/lib.rs]: ok, I thought that by this we can show that we do care for other stuff than just plain tokio. :) And mybe for better searchability in serach engines :)

**Review by tworec — APPROVED** [review]
lgtm :)

**Comment by palfrey** [comment]
Pushed out as release 0.5.0

**Comment by tworec** [comment]
thx. I'm using it 
https://github.com/golemfactory/yagna/blob/b35b454565d1bbc53b9e66f985c5c0442fca11a2/core/market/decentralized/Cargo.toml#L59

**Pull request merged**